File: server/datasets/java_source-code/Multiple-A3/organic_9.java
// Linked List Reversal - organic submission 9
// Adds insertAtHead alongside append, reverses using while-true/break.

class Node {
    int data;
    Node next;

    public Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;

    public LinkedList() {
        this.head = null;
    }

    public void insertAtHead(int data) {
        Node n = new Node(data);
        n.next = this.head;
        this.head = n;
    }

    public void append(int data) {
        Node n = new Node(data);
        if (this.head == null) {
            this.head = n;
            return;
        }
        Node walker = this.head;
        while (true) {
            if (walker.next == null) {
                break;
            }
            walker = walker.next;
        }
        walker.next = n;
    }

    public void reverse() {
        Node previous = null;
        Node current = this.head;
        while (true) {
            if (current == null) {
                break;
            }
            Node nxt = current.next;
            current.next = previous;
            previous = current;
            current = nxt;
        }
        this.head = previous;
    }

    public void printValues() {
        Node walker = this.head;
        StringBuilder sb = new StringBuilder();
        while (walker != null) {
            sb.append(walker.data);
            if (walker.next != null) sb.append(" ");
            walker = walker.next;
        }
        System.out.println(sb.toString());
    }

    public static void main(String[] args) {
        LinkedList ll = new LinkedList();
        for (int v : new int[]{3, 1, 4, 1, 5, 9}) {
            ll.append(v);
        }
        ll.insertAtHead(0);
        ll.printValues();
        ll.reverse();
        ll.printValues();
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_a3/organic_9.py
# Linked List Reversal - organic submission 9
# Adds insert_at_head alongside append, and reverses using enumerate-style
# manual walking with a while-True/break structure.

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def insert_at_head(self, data):
        n = Node(data)
        n.next = self.head
        self.head = n

    def append(self, data):
        n = Node(data)
        if self.head is None:
            self.head = n
            return
        walker = self.head
        while True:
            if walker.next is None:
                break
            walker = walker.next
        walker.next = n

    def reverse(self):
        previous = None
        current = self.head
        while True:
            if current is None:
                break
            nxt = current.next
            current.next = previous
            previous = current
            current = nxt
        self.head = previous

    def values(self):
        out = []
        walker = self.head
        while walker is not None:
            out.append(walker.data)
            walker = walker.next
        return out


if __name__ == "__main__":
    ll = LinkedList()
    for v in [3, 1, 4, 1, 5, 9]:
        ll.append(v)
    ll.insert_at_head(0)
    print(ll.values())
    ll.reverse()
    print(ll.values())
